sky/backends/backend_utils.py

```diff
@@ -2276,19 +2300,45 @@ def is_controller_up(
         controller_status, handle = None, None
         if record is not None:
             controller_status, handle = record['status'], record['handle']
+            # We check the connection even if the cluster has a cached status UP
+            # to make sure the controller is actually accessible, as the cached
+            # status might be stale.
+            need_connection_check = True
+
+    error_msg = None
+    if controller_status == status_lib.ClusterStatus.STOPPED:
+        error_msg = stopped_message
+    elif controller_status is None or handle.head_ip is None:
```

Comment:
Worth adding comment https://github.com/skypilot-org/skypilot/pull/3288#discussion_r1518944312 to inline?